pipeline/customer/customer.py

```diff
@@ -69,17 +74,18 @@ def process(self, element, *args, **kwargs):
             customer_detail = {}
         if not customer_orders:
             customer_orders = {}
-        yield json.dumps({
+        output = {
             'id': customer_id,
             'first_name': customer_detail.get(detail.Field.FirstName),
             'last_name': customer_detail.get(detail.Field.LastName),
-            'sum': customer_orders.get(order.Field.Sum),
+            self.aggregation.get(): customer_orders.get(self.aggregation.get()),
             'errors': entry.get(Field.Errors),
-        }, sort_keys=True, cls=common.PipelineJSONEncoder)
+        }
+        yield json.dumps(output, sort_keys=True, cls=common.PipelineJSONEncoder)
 
 
 def run(argv=None):
-    parser = argparse.ArgumentParser()
+    parser = _BeamArgumentParser()
```

Comment:
You will probably have to disable a pylon warning here (use of private field).

Reply:
I had to create a class now, because otherwise the value provider input does not work. That code here is gone now.

Reply:
I would be interested in hearing more next week. This could be interesting to make our pipelines compatible with Apache Flink.